pytest_tests/tests/acl/test_eacl_filters.py

```diff
@@ -627,3 +657,556 @@ def test_extended_acl_allow_filters_object(
+        user_wallet = wallets.get_wallet()
+
+        cid = eacl_container
+        oid = None
+
+        epoch = self.get_epoch()
+
+        with allure.step(f"Set EACLs for GET/PUT to restrict operations with expiration attribute"):
+            eacl_deny = [
+                EACLRule(
+                    access=EACLAccess.DENY,
+                    role=EACLRole.USER,
+                    filters=EACLFilters(
+                        [
+                            EACLFilter(
+                                header_type=EACLHeaderType.OBJECT,
+                                match_type=EACLMatchType.NUM_GE,
+                                key=self.EXPIRATION_OBJECT_ATTR,
+                                value=epoch + 2,
+                            ),
+                        ]
+                    ),
+                    operation=EACLOperation.GET,
+                ),
+                EACLRule(
+                    access=EACLAccess.DENY,
+                    role=EACLRole.USER,
+                    filters=EACLFilters(
+                        [
+                            EACLFilter(
+                                header_type=EACLHeaderType.OBJECT,
+                                match_type=EACLMatchType.NUM_GT,
+                                key=self.EXPIRATION_OBJECT_ATTR,
+                                value=epoch + 2,
+                            ),
+                        ]
+                    ),
+                    operation=EACLOperation.PUT,
+                ),
+                EACLRule(
+                    access=EACLAccess.DENY,
+                    role=EACLRole.USER,
+                    filters=EACLFilters(
+                        [
+                            EACLFilter(
+                                header_type=EACLHeaderType.OBJECT,
+                                match_type=EACLMatchType.NUM_LT,
+                                key=self.EXPIRATION_OBJECT_ATTR,
+                                value=epoch + 2,
+                            ),
+                        ]
+                    ),
+                    operation=EACLOperation.PUT,
+                ),
+            ]
+            set_eacl(
+                user_wallet.wallet_path,
+                cid,
+                create_eacl(cid, eacl_deny, shell=self.shell),
+                shell=self.shell,
+                endpoint=self.neofs_env.sn_rpc,
+            )
+            wait_for_cache_expired()
+            get_eacl(
+                user_wallet.wallet_path,
+                cid,
+                shell=self.shell,
+                endpoint=self.neofs_env.sn_rpc,
+            )
+
+        with allure.step("Add test object to container"):
+            file_path = generate_file(complex_object_size)
+
+            oid = put_object_to_random_node(
+                user_wallet.wallet_path,
+                file_path,
+                cid,
+                shell=self.shell,
+                neofs_env=self.neofs_env,
+                lifetime=2,
```

Comment:
I think you should use `expire-at`, lifetime is relative to the current epoch which technically can change during the test.

Reply:
fixed